Tella/Data/MainAppModel.swift

```diff
@@ -76,6 +76,8 @@ class MainAppModel: ObservableObject, AppModelFileManagerProtocol {
     
     init(networkMonitor:NetworkMonitor) {
         self.networkMonitor = networkMonitor
+        // To initialize the NetworkMonitor singleton class
+        _ = NetworkMonitor.shared.isConnected
```

Comment:
You don't need to reinitiate the `NetworkMonitor` here, you can update it in `AppViewState` 
`   let networkMonitor = NetworkMonitor.shared.    
`

Reply:
Code moved to `AppViewState `